src/components/views/rooms/MemberInfo.js

```diff
@@ -92,75 +101,69 @@ module.exports = React.createClass({
-        const userIds = [
-            this.props.member.userId,
-            MatrixClientPeg.get().credentials.userId
-        ];
-        let existingRoomId = null;
-        let invitedRoomId = null;
+    onDeviceVerificationChanged: function(userId, device) {
+        if (!this._enableDevices) {
+            return;
+        }
 
-        // roomId can be null here because of a hack in MatrixChat.onUserClick where we
-        // abuse this to view users rather than room members.
-        let currentMembers;
-        if (this.props.member.roomId) {
-            const currentRoom = MatrixClientPeg.get().getRoom(this.props.member.roomId);
-            currentMembers = currentRoom.getJoinedMembers();
+        if (userId == this.props.member.userId) {
+            // no need to re-download the whole thing; just update our copy of
+            // the list.
+            var devices = MatrixClientPeg.get().getStoredDevicesForUser(userId);
+            this.setState({devices: devices});
         }
+    },
 
-        // reuse the first private 1:1 we find
-        existingRoomId = null;
+    onRoom: function(room) {
+        this.forceUpdate();
+    },
 
-        for (let i = 0; i < rooms.length; i++) {
-            // don't try to reuse public 1:1 rooms
-            const join_rules = rooms[i].currentState.getStateEvents("m.room.join_rules", '');
-            if (join_rules && join_rules.getContent().join_rule === 'public') continue;
+    onDeleteRoom: function(roomId) {
+        this.forceUpdate();
+    },
 
-            const members = rooms[i].getJoinedMembers();
-            if (members.length === 2 &&
-                userIds.indexOf(members[0].userId) !== -1 &&
-                userIds.indexOf(members[1].userId) !== -1)
-            {
-                existingRoomId = rooms[i].roomId;
-                break;
-            }
+    onRoomTimeline: function(ev, room, toStartOfTimeline) {
+        if (toStartOfTimeline) return;
+        this.forceUpdate();
+    },
 
-            const invited = rooms[i].getMembersWithMembership('invite');
-            if (members.length === 1 &&
-                invited.length === 1 &&
-                userIds.indexOf(members[0].userId) !== -1 &&
-                userIds.indexOf(invited[0].userId) !== -1 &&
-                invitedRoomId === null)
-            {
-                invitedRoomId = rooms[i].roomId;
-                // keep looking: we'll use this one if there's nothing better
-            }
-        }
+    onRoomName: function(room) {
+        this.forceUpdate();
+    },
 
-        if (existingRoomId === null) {
-            existingRoomId = invitedRoomId;
+    onRoomReceipt: function(receiptEvent, room) {
+        // because if we read a notification, it will affect notification count
+        // only bother updating if there's a receipt from us
+        if (Receipt.findReadReceiptFromUserId(receiptEvent, MatrixClientPeg.get().credentials.userId)) {
+            this.forceUpdate();
```

Comment:
It doesn't really, it's just the same optimisation used in RoomList that we can skip updating if the read receipt is not from us because we only show the read count.